.github/workflows/build-domain.yaml

```diff
@@ -0,0 +1,288 @@
+name: Build domain
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      domains: ${{ steps.define-matrices.outputs.domains }}
+      projects: ${{ steps.define-matrices.outputs.projects }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Validate required domain structure
+        shell: bash
+        run: |
+          function required_folder {
+            [ -d "$1" ] || { printf '::error title=Missing folder::%s\n' "$1"; exit 1; }
+          }
+
+          function required_file {
+            [ -f "$1" ] || { printf '::error title=Missing file::%s\n' "$1"; exit 1; }
+          }
+
+          find domains -mindepth 1 -maxdepth 1 -type d | while read domain; do
+            required_folder "$domain/.devcontainer"
+            required_file "$domain/.devcontainer/Dockerfile"
+            required_file "$domain/.devcontainer/devcontainer.json"
+
+            required_folder "$domain/.vscode"
+            required_file "$domain/.vscode/settings.json"
+            required_file "$domain/.vscode/extensions.json"
+
+            required_file "$domain"/*.sln
```

Comment:
This is just a summary of what we talked about in person: These files and folders are technically not required for building the domain. They are listed here as required to ensure that each domain provide a working setup for development.
@CodeReaper: Please mark this conversation as resolved if you agree.